pkg/agent/netpol/netpol.go

```diff
@@ -55,6 +60,30 @@ func Run(ctx context.Context, nodeConfig *config.Node) error {
 		return err
 	}
 
+	// As kube-router netpol requires addresses to be available in the node object
+	// Wait until the node has ready addresses to avoid race conditions (max 1 minute).
+	// TODO: Replace with non-deprecated PollUntilContextTimeout when our and Kubernetes code migrate to it
+	wait.PollImmediateWithContext(ctx, 2*time.Second, 60*time.Second, func(ctx context.Context) (bool, error) {
+		taintFound := false
+		// Get the node object
+		node, err := client.CoreV1().Nodes().Get(context.TODO(), nodeConfig.AgentConfig.NodeName, metav1.GetOptions{})
+		if err != nil {
+			logrus.Errorf("Error getting the node object: %v", err)
+			return false, err
+		}
+		// Check for the uninitialized taint that should be removed by cloud-provider
+		// If there is no cloud-provider, the tain will not be there
+		for _, taint := range node.Spec.Taints {
+			if taint.Key == cloudproviderapi.TaintExternalCloudProvider {
+				taintFound = true
+			}
+		}
+		if !taintFound {
+			return true, nil
+		}
+		return false, nil
```

Comment:
You don't need an intermediate var, just return false as soon as you find it, and true if you don't.
```suggestion
				return false, nil
			}
		}
		return true, nil
```

Reply:
thanks Brad, you are totally right